src/StackExchange.Redis/Interfaces/IDatabaseAsync.cs

```diff
@@ -312,6 +312,79 @@ public interface IDatabaseAsync : IRedisAsync
+        /// <param name="expiry">The timeout to set.</param>
+        /// <param name="when">under which condition the expiration will be set using <see cref="ExpireWhen"/>.</param>
+        /// <param name="flags">The flags to use for this operation.</param>
+        /// <returns> Empty array if the key does not exist. Otherwise returns an array where each item is the result of operation for given fields;
+        ///		2: field deleted because the specified expiration time is due
+        ///		1: expiration time set/updated
+        /// 	0: expiration time is not set/update (a specified ExpireWhen condition is not met)
+        ///		-2 : no such field
+        /// </returns>
+        Task<ExpireResult[]> HashFieldExpireAsync(RedisKey key, RedisValue[] hashFields, TimeSpan expiry, ExpireWhen when = ExpireWhen.Always, CommandFlags flags = CommandFlags.None);
+
+        /// <summary>
+        ///	Set the time out on a field of the given set of fields of hash
+        /// After the timeout has expired, the field of the hash will automatically be deleted.
+        /// </summary>
+        /// <param name="key">The key of the hash.</param>
+        /// <param name="hashFields">The fields in the hash to set expire time.</param>
+        /// <param name="expiry">The exact date to expiry to set.</param>
+        /// <param name="when">under which condition the expiration will be set using <see cref="ExpireWhen"/>.</param>
+        /// <param name="flags">The flags to use for this operation.</param>
+        /// <returns> Empty array if the key does not exist. Otherwise returns an array where each item is the result of operation for given fields;
+        ///		2: field deleted because the specified expiration time is due
+        ///		1: expiration time set/updated
+        /// 	0: expiration time is not set/update (a specified ExpireWhen condition is not met)
+        ///		-2 : no such field
+        /// </returns>
+        Task<ExpireResult[]> HashFieldExpireAsync(RedisKey key, RedisValue[] hashFields, DateTime expiry, ExpireWhen when = ExpireWhen.Always, CommandFlags flags = CommandFlags.None);
+
+        /// <summary>
+        /// For each specified field, it gets the expiration time as a Unix timestamp in milliseconds (milliseconds since the Unix epoch)
+        /// </summary>
+        /// <param name="key">The key of the hash.</param>
+        /// <param name="hashFields">The fields in the hash to get expire time.</param>
+        /// <param name="flags">The flags to use for this operation.</param>
+        /// <returns>Empty array if the key does not exist. Otherwise returns the result of operation for given fields
+        ///     expiration time: as a UNIX timestamp in milliseconds
+        ///     -1: if field has no associated expiration time 
+        ///     -2: no such field
+        /// </returns>
+        Task<long[]> HashFieldGetExpireDateTimeAsync(RedisKey key, RedisValue[] hashFields, CommandFlags flags = CommandFlags.None);
```

Comment:
Normally we'd want to parse this out to `TimeSpan` here, but I guess we can't do that conveniently here because of the alternative (`-1`/`-2`) results; I think it's probably fine, all things considered - any concerns @NickCraver  ?

Reply:
Yeah I think we're in magical territory there with TimeSpan, unfortunately - best option is as-is.